partiql-ast/src/main/resources/partiql_ast.ion

```diff
@@ -0,0 +1,624 @@
+        wildcard::{},
+        unpivot::{},
+      ],
+    ],
+  },
+
+  // Scalar Function Call — for now, `function` is just a single identifier
+  call::{
+    function: string,
+    args:     list::[expr],
+  },
+
+  // Aggregate Function Call
+  agg::{
+    function:   string,
+    args:       list::[expr],
+    quantifier: set_quantifier,
+  },
+
+  // A parameter `?`
+  parameter::{
+    index: int,
+  },
+
+  // Unary Operators
+  unary::{
+    op:   [ NOT, POS, NEG ],
+    expr: expr,
+  },
+
+  // Binary Operators
+  binary::{
+    op: [
+      PLUS, MINUS, TIMES, DIVIDE, MODULO, CONCAT,
+      AND, OR,
+      EQ, NE, GT, GTE, LT, LTE
+    ],
+    lhs: expr,
+    rhs: expr,
+  },
+
+  // Collection Constructors
+  collection::{
+    type:   [ BAG, ARRAY, LIST, SEXP ], // syntax `<< >>`, `[ ]`, `[VALUES|LIST] ( )`, `SEXP ( )`
+    values: list::[expr],
+  },
+
+  // Tuple Constructor
+  tuple::{
+    fields: list::[field],
+    _: [
+      field::{
+        name:   expr,
+        value:  expr,
+      },
+    ],
+  },
+
+  // Date Literal
+  date::{
+    year:   long,
+    month:  long,
+    day:    long,
+  },
+
+  // Time Literal
+  time::{
+    hour:               long,
+    minute:             long,
+    second:             long,
+    nano:               long,
+    precision:          long,
+    with_tz:            bool,
+    tz_offset_minutes:  optional::long,
+  },
+
+  // The `like` SQL special form
+  like::{
+    value:    expr,
+    pattern:  expr,
+    escape:   optional::expr,
+  },
+
+  // The `between` SQL special form
+  between::{
+    value:  expr,
+    from:   expr,
+    to:     expr,
+  },
+
+  // The `in` SQL special form
+  in_collection::{
+    lhs: expr,
+    rhs: expr,
+  },
+
+  // The `is` SQL special form
+  is_type::{
+    value:  expr,
+    type:   '.type',
+  },
+
+  // The simple and searched `case` switch SQL special form F261-01, F261-02
+  switch::{
+    expr:     optional::expr,
+    branches: list::[branch],
+    default:  optional::expr,
+    _: [
+      branch::{
+        condition:  expr,
+        expr:       expr,
+      },
+    ],
+  },
+
+  // The `coalesce` SQL special form F261-04
+  coalesce::{
+    args: list::[expr],
+  },
+
+  // The `nullif` SQL special form F261-03
+  null_if::{
+    expr0: expr,
+    expr1: expr,
+  },
+
+  // The `cast` SQL special form F201
+  cast::{
+    value:    expr,
+    as_type:  '.type',
+  },
+
+  // The `can_cast` PartiQL special form
+  can_cast::{
+    value:    expr,
+    as_type:  '.type',
+  },
+
+  // The `can_lossless_cast` PartiQL special form
+  can_lossless_cast::{
+    value:    expr,
+    as_type:  '.type',
+  },
+
+  // The [INNER|OUTER] [UNION|INTERSECT|EXCEPT] PartiQL multiset special forms
+  set::{
+    op:         [ UNION, INTERSECT, EXCEPT ],
+    quantifier: set_quantifier,
+    outer:      bool,
+    lhs:        expr,
+    rhs:        expr,
+  },
+
+  // The PartiQL `sfw` query expression
+  s_f_w::{
+    select:   select,
```

Comment:
That's very similar to how this PR models it. It is actually wrong however because this does not consider UNION / INTERSECT / EXCEPT clauses which have been missing from PartiQL hence the simplified model of a SFW. We really need to separate the ORDER / FETCH clauses.

I will address that now.